src/org/pcic/util/Utils.groovy

```diff
@@ -0,0 +1,70 @@
+package org.pcic.util
+
+import org.pcic.GitUtils
+import org.pcic.DockerUtils
+
+
+class Utils implements Serializable {
+    def steps
+
+    Utils(steps) {
+        this.steps = steps
+    }
+
+    /**
+     * This method takes in a list of keys and a map, and returns a map with
+     * updates values.
+     */
+    Map getUpdatedArgs(ArrayList keys, Map argsToUpdate) {
+        return updateArgs(getDefaultArgs(keys), argsToUpdate)
+    }
+
+    /**
+     * Returns a submap given a list of keys
```

Comment:
Suggest rewrite:

Returns a map of default arguments, given a list of keys. Default arguments are common to many steps and classes, so are defined once here. Because only a subset of the arguments is required or valid for any given step, the keys of those arguments are passed in and only those defaults are returned.

Reply:
See note above regarding purpose vs. method.